### PR TITLE
 Adding stack to errors

### PR DESCRIPTION
**Pull request by wazsone**
## Description
Make sure that all errors have a stack and message 

Before:
```
Running tool: /usr/local/go/bin/go test -timeout 30s -run ^TestExcludedPrefixesClient_Close_PrefixesAreRemoved$ github.com/networkservicemesh/sdk/pkg/networkservice/common/excludedprefixes

2023/02/01 06:58:13 [ERROR] [ExcludedPrefixesClient:Request] Source or destination IPs are overlapping with excluded prefixes, srcIPs: [a.16.0.100/32], dstIPs: [b.16.0.103/32], excluded prefixes: [], error: invalid CIDR address: a.16.0.100/32
--- FAIL: TestExcludedPrefixesClient_Close_PrefixesAreRemoved (0.00s)
    /home/ubuntu/nsm/wazsone/sdk/pkg/networkservice/common/excludedprefixes/client_test.go:153: 
        	Error Trace:	/home/ubuntu/nsm/wazsone/sdk/pkg/networkservice/common/excludedprefixes/client_test.go:153
        	Error:      	Received unexpected error:
        	            	invalid CIDR address: a.16.0.100/32
        	Test:       	TestExcludedPrefixesClient_Close_PrefixesAreRemoved
FAIL
FAIL	github.com/networkservicemesh/sdk/pkg/networkservice/common/excludedprefixes	0.016s
FAIL
```
After:
```
Running tool: /usr/local/go/bin/go test -timeout 30s -run ^TestExcludedPrefixesClient_Close_PrefixesAreRemoved$ github.com/networkservicemesh/sdk/pkg/networkservice/common/excludedprefixes

2023/02/01 06:59:29 [ERROR] [ExcludedPrefixesClient:Request] Source or destination IPs are overlapping with excluded prefixes, srcIPs: [a.16.0.100/32], dstIPs: [b.16.0.103/32], excluded prefixes: [], error: failed to parse a.16.0.100/32 as CIDR: invalid CIDR address: a.16.0.100/32
--- FAIL: TestExcludedPrefixesClient_Close_PrefixesAreRemoved (0.00s)
    /home/ubuntu/nsm/wazsone/sdk/pkg/networkservice/common/excludedprefixes/client_test.go:153: 
        	Error Trace:	/home/ubuntu/nsm/wazsone/sdk/pkg/networkservice/common/excludedprefixes/client_test.go:153
        	Error:      	Received unexpected error:
        	            	invalid CIDR address: a.16.0.100/32
        	            	failed to parse a.16.0.100/32 as CIDR
        	            	github.com/networkservicemesh/sdk/pkg/networkservice/common/excludedprefixes.validateIPs
        	            		/home/ubuntu/nsm/wazsone/sdk/pkg/networkservice/common/excludedprefixes/client.go:230
        	            	github.com/networkservicemesh/sdk/pkg/networkservice/common/excludedprefixes.(*excludedPrefixesClient).Request
        	            		/home/ubuntu/nsm/wazsone/sdk/pkg/networkservice/common/excludedprefixes/client.go:123
        	            	github.com/networkservicemesh/sdk/pkg/networkservice/common/excludedprefixes_test.TestExcludedPrefixesClient_Close_PrefixesAreRemoved
        	            		/home/ubuntu/nsm/wazsone/sdk/pkg/networkservice/common/excludedprefixes/client_test.go:152
        	            	testing.tRunner
        	            		/usr/local/go/src/testing/testing.go:1439
        	            	runtime.goexit
        	            		/usr/local/go/src/runtime/asm_amd64.s:1571
        	Test:       	TestExcludedPrefixesClient_Close_PrefixesAreRemoved
FAIL
FAIL	github.com/networkservicemesh/sdk/pkg/networkservice/common/excludedprefixes	0.014s
FAIL
```

## Issue link
https://github.com/networkservicemesh/sdk/issues/705

## How Has This Been Tested?

- [ ] Added unit testing to cover
- [x] Tested manually
- [ ] Tested by integration testing
- [ ] Have not tested

## Types of changes

- [ ] Bug fix
- [ ] New functionality
- [ ] Documentation
- [x] Refactoring
- [ ] CI

### CODE REVIEW

File: pkg/networkservice/utils/checks/checkconnection/client.go [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// Copyright (c) 2020-2022 Cisco and/or its affiliates.
+// Copyright (c) 2020-2023 Cisco and/or its affiliates.
```
Comment (glazychev-art) [inline pkg/networkservice/utils/checks/checkconnection/client.go:1]: unnecessary change

File: pkg/networkservice/core/eventchannel/monitor_connection_client.go [hunk 2]
```diff
@@ -47,7 +47,7 @@ func NewMonitorConnectionMonitorConnectionsClient(ctx context.Context, eventCh <
 func (m *monitorConnectionMonitorConnectionsClient) Recv() (*networkservice.ConnectionEvent, error) {
 	select {
 	case <-m.ctx.Done():
-		return nil, m.ctx.Err()
+		return nil, errors.WithStack(m.ctx.Err())
```
Comment (glazychev-art) [inline pkg/networkservice/core/eventchannel/monitor_connection_client.go:50]: I think it should be "application context is done"
(and other places like `m.ctx.Err()`)

**Review by glazychev-art — COMMENTED** [review]
In general it looks good to me.
Left a couple of comments

File: pkg/registry/common/memory/ns_server.go [hunk 3]
```diff
@@ -103,7 +106,7 @@ func (s *memoryNSServer) Find(query *registry.NetworkServiceQuery, server regist
 	var err error
 	for ; err == nil; err = s.receiveEvent(query, server, eventCh) {
 	}
-	if err != io.EOF {
+	if errors.Cause(err).Error() != io.EOF.Error() {
```
Comment (denis-tingaikin) [inline pkg/registry/common/memory/ns_server.go:109]: Please apply this for other places.

```suggestion
	if errors.Is(err, io.EOF) {
```

File: pkg/registry/common/memory/nse_server.go [hunk 4]
```diff
@@ -106,7 +109,7 @@ func (s *memoryNSEServer) Find(query *registry.NetworkServiceEndpointQuery, serv
 	var err error
 	for ; err == nil; err = s.receiveEvent(query, server, eventCh) {
 	}
-	if err != io.EOF {
+	if errors.Cause(err).Error() != io.EOF.Error() {
```
Comment (denis-tingaikin) [inline pkg/registry/common/memory/nse_server.go:112]: Same

File: pkg/registry/common/memory/nse_server.go [hunk 5]
```diff
@@ -146,14 +149,14 @@ func (s *memoryNSEServer) receiveEvent(
...
 			if err := server.Send(event); err != nil {
 				if server.Context().Err() != nil {
-					return io.EOF
+					return errors.Wrapf(io.EOF, "find context has a error %s", server.Context().Err().Error())
 				}
-				return err
+				return errors.Wrapf(err, "NetworkServiceRegistry find server failed to send a response %s", event.String())
 			}
 		}
 		return nil
```
Comment (denis-tingaikin) [inline pkg/registry/common/memory/nse_server.go:162]: Note: EOF is fine answer for the memory server.

File: pkg/registry/common/memory/ns_server.go [hunk 6]
```diff
@@ -143,7 +146,7 @@ func (s *memoryNSServer) receiveEvent(
 ) error {
 	select {
 	case <-server.Context().Done():
-		return io.EOF
+		return errors.Wrap(io.EOF, "find context is done")
```
Comment (denis-tingaikin) [inline pkg/registry/common/memory/ns_server.go:149]: EOF is fine answer for the server

File: pkg/registry/common/memory/ns_server.go [hunk 7]
```diff
@@ -152,9 +155,9 @@ func (s *memoryNSServer) receiveEvent(
 
 			if err := server.Send(nse); err != nil {
 				if server.Context().Err() != nil {
-					return io.EOF
+					return errors.Wrapf(io.EOF, "find context has a error %s", server.Context().Err().Error())
```
Comment (denis-tingaikin) [inline pkg/registry/common/memory/ns_server.go:158]: EOF is fine answer for the server

**Review by denis-tingaikin — COMMENTED** [review]
1.  use errors.Is() for places where we need to compare errors 
2. EOF is fine error for memory nse/ns servers

**Review by glazychev-art — APPROVED** [review]

**Review by denis-tingaikin — APPROVED** [review]
